packages/component-library/draft/Kaizen/Modal/Modal.elm

```diff
@@ -131,33 +143,56 @@ view (Config config) =
             []
         , case config.variant of
             Generic content size ->
-                GenericModal.view
-                    [ modalBox content size ]
-                    (genericModal |> GenericModal.events genericModalEvents)
+                GenericModal.view (GenericModal.Custom size)
+                    content
+                    (genericModalConfig |> GenericModal.events genericModalEvents)
+
+            Confirmation confirmationType configs ->
+                let
+                    withOnDismiss confirmationConfig =
```

Comment:
what do you think about extracting this to external functions to be piped like `confirmLabel`, `dismissLabel`, etc?

Reply:
`ConfirmationModal.elm` does have those functions already and Modal.elm does in fact call them. This config is more so a contract that `Modal.elm` needs for the variant. So when a consumer asks for a Confirmation variant from `Modal.elm` this config is the required data Modal needs in order to use `ConfirmationModal.elm`. You could call `ConfirmationModal.elm` directly if you wanted to build some sort of hybrid modal and use those functions you are talking about. We want to discourage people from doing that though.